index.js

```diff
@@ -85,11 +85,11 @@ function compression (options) {
       }
 
       return stream
-        ? stream.write(toBuffer(chunk, encoding))
+        ? stream.write(toBuffer(chunk, encoding), cb || encoding)
         : _write.call(this, chunk, encoding)
     }
 
-    res.end = function end (chunk, encoding) {
+    res.end = function end (chunk, encoding, cb) {
```

Comment:
Looking at https://nodejs.org/dist/latest-v18.x/docs/api/http.html#responseenddata-encoding-callback it seems that the first argument can also be a callback, so `res.end(cb)` should work too I guess.